packages/teams-js/src/public/sharing.ts

```diff
@@ -55,103 +70,88 @@ export namespace sharing {
    *
    * @internal
    */
+  export function shareWebContent(
+    shareWebContentRequest: IShareRequest<IShareRequestContentType>,
+    callback: (err?: SdkError) => void,
+  ): void;
   export function shareWebContent(
     shareWebContentRequest: IShareRequest<IShareRequestContentType>,
     callback?: (err?: SdkError) => void,
-  ): void {
-    if (!validateNonEmptyContent(shareWebContentRequest, callback)) {
-      return;
-    }
-
-    if (!validateTypeConsistency(shareWebContentRequest, callback)) {
-      return;
-    }
-
-    if (!validateContentForSupportedTypes(shareWebContentRequest, callback)) {
-      return;
+  ): Promise<void> {
+    // validate the given input (synchronous check)
+    try {
+      validateNonEmptyContent(shareWebContentRequest);
+      validateTypeConsistency(shareWebContentRequest);
+      validateContentForSupportedTypes(shareWebContentRequest);
+    } catch (err) {
+      //return the error via callback(v1) or rejected promise(v2)
+      const wrappedFunction: InputFunction<void> = () => new Promise((resolve, reject) => reject(err));
```

Comment:
is `() => new Promise((resolve, reject) => reject(err));` the same as `() => Promise.reject(err)`? If so, I think I would prefer that second version.

Reply:
Done